packages/build/src/log/messages/compatibility.js

```diff
@@ -176,3 +213,50 @@ const getIncompatiblePlugin = function ({
 const getVersionedPackage = function (packageName, version = '') {
   return version === '' ? '' : `@${version}`
 }
+
+// Print an error message when unsupported versions of plugins are used.
+// This can only happen when they are installed to `package.json`.
+export const logUnsupportedPluginVersions = function (logs, pluginsOptions) {
+  const unsupportedVersions = pluginsOptions.filter(hasUnsupportedVersion).map(getExcludedVersion)
+  if (unsupportedVersions.length === 0) {
+    return
+  }
+
+  logErrorHeader(logs, 'Unsupported plugin version(s)')
+  logErrorArray(logs, unsupportedVersions)
+
+  const error = new Error('Unsupported plugin version(s) detected')
+  addErrorInfo(error, { type: 'pluginUnsupportedVersion' })
+  throw error
+}
+
+const hasUnsupportedVersion = function ({ pluginPackageJson: { version }, manifest: { excludedVersionsRange } }) {
+  // https://github.com/npm/node-semver#hyphen-ranges-xyz---abc
+  // excludedVersionsRange is a range of versions that are not supported e.g 4.0.0 - 4.25.0
+  // semver hyphen range is inclusive 4.0.0 - 4.25.0 is same as >= 4.0.0 || < 4.26.0;
+  return (
+    version !== undefined && excludedVersionsRange !== undefined && semver.satisfies(version, excludedVersionsRange)
```

Comment:
```suggestion
    version !== undefined && excludedVersionsRange !== undefined && semver.satisfies(version, excludedVersionsRange, { includePrerelease: true })
```
We need to allow prerelease versions. By default prereleases never satisfy semver